File: stock_buy_and_sell.py
def maxProfit(Prices):
    max_profit=0
    min_price=Prices[0]
    n=len(Prices)
    for i in range(1,n):
        profit=Prices[i]-min_price
        min_price=min(min_price,Prices[i])
        max_profit=max(profit,max_profit)

    return max_profit

def main():
    n=int(input("Enter number of days : "))
    Prices = list(map(int, input(f"Enter prices for {n} days: ").split()))

    if len(Prices)!=n:
        print("Number of prices entered does not match n!")
        return
    result=maxProfit(Prices)
    print("Maximum Profit : ",result)

if __name__ == "__main__":
        main()

File: stock_buy_and_sell.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int n = prices.size();
        int profit, min_price = prices[0];
        for(int i = 1; i < n; i++) {
            profit = prices[i] - min_price;
            min_price = min(min_price, prices[i]);
            max_profit = max(max_profit, profit);
        }
        return max_profit;
    }
int main() {
    int n;
    cout << "Enter number of days: ";
    cin >> n;

    vector<int> prices(n);
    cout << "Enter prices for " << n << " days: ";
    for(int i = 0; i < n; i++) {
        cin >> prices[i];
    }

    int result =maxProfit(prices);
    cout << "Maximum Profit: " << result << endl;

    return 0;
}